flocker/node/testtools.py

```diff
@@ -145,14 +146,9 @@ def test_calculate_necessary_state_changes(self):
             ``IStateChange`` provider.
             """
             deployer = fixture(self)
-            d = deployer.discover_local_state(NodeState(hostname=u"127.0.0.1"))
```

Comment:
You should add a test for `discover_state`, in which case the extra test you wrote for `P2PNodeDeployer` isn't necessary.